src/tests/ffi.cpp

```diff
@@ -5923,3 +5925,36 @@ TEST_F(rnp_tests, test_ffi_security_profile)
 
     rnp_ffi_destroy(ffi);
 }
+
+TEST_F(rnp_tests, test_result_to_string)
+{
+    const char *          result_string = NULL;
+    rnp_result_t          code;
+    std::set<std::string> stringset;
+
+    result_string = rnp_result_to_string(RNP_SUCCESS);
+    assert_string_equal(result_string, "Success");
+
+    /* Cover all defined error code ranges,
+     * check that each defined
+     * code has corresponding unique string */
+
+    std::vector<std::pair<rnp_result_t, rnp_result_t>> error_codes = {
+      {RNP_ERROR_GENERIC, RNP_ERROR_NULL_POINTER},
+      {RNP_ERROR_ACCESS, RNP_ERROR_WRITE},
+      {RNP_ERROR_BAD_STATE, RNP_ERROR_SIGNATURE_UNKNOWN},
+      {RNP_ERROR_NOT_ENOUGH_DATA, RNP_ERROR_EOF}};
```

Comment:
I'm afraid this is not going to catch the return of this bug, as the new codes will be added after the range ends, as those at the end of the range are not special sentinel entries but actual error codes.

One way I see to nail it is parsing rnp_err.h and generating a test program *at testing time* to exercise each defined error code.

Reply:
Parsing of the `rnp_err.h` seem to be an overkill, based on possible impact of this issue.